Add unique_id to RFLink entities

javicalle opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add `unique_id` property to the RFLink entities.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->
There is no changes to device settings
```yaml
# Example configuration.yaml entry
switch:
  - platform: rflink
    device_defaults:
      fire_event: true
      signal_repetitions: 2
    devices:
      newkaku_0000c6c2_1:
        name: Ceiling fan
      conrad_00785c_0a:
        name: Motion sensor kitchen
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->
This change has been successfully tested in my local environment, including automations, HA interactions and phisical devices interactions.


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

javicalle committed:
add unique_id to RFLink entities

javicalle committed:
add unique_id to RFLink entities

javicalle committed:
entity_id bug

javicalle committed:
black format

javicalle committed:
black format

javicalle committed:
fix coverage check

frenck commented:
Hi there @javicalle!

Thanks for opening up a PR! However, there are all kinds of unrelated commits in this PR. Could you please clean that up so the PR only contains your changes?

Thanks! 👍

javicalle commented:
I think I was able to resolve the mess I had made with commits.
I hope the tests go fine

javicalle commented:
Local coverage ups to 100%
Not sure why codecov fails now. Is it a problem?

javicalle committed:
device to entity rename

javicalle commented:
I've been reviewing how the `unique_id` has been implemented in other integrations. 
In general, they follow the [recommendations of the documentation in this regard](https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements) and those cases that are not, but there are possibilities, it has been rejected (#34668).

So, what is the situation in this case?
(disclaimer: the following is my personal analysis of RFLink library and may be incorrect and / or totally wrong)
The RFLink gateway reads RF signals. These signals have limited information on the device that generates them (besides the signal value) they have 'protocol', 'id' and 'button'. There are represented in the format `<protocol> _ <id> _ <button>` and is the value stored as `_device_id`

And, are these values uniques? 
It cannot be guaranteed. In fact, some devices have a switch that allows you to change the ID to avoid conflicts, or there are others that change the ID every time you replace the batteries. 
So it may not seem like a good candidate to define the `unique_id` value.

BUT IMHO there is another approach that is intended to follow this PR.
In the configuration you define an entity with a entity_id. And the RFLink library allows you to control changes to the `device_id` by defining device aliases. So you can have a device configured that is 'unique' in the configuration.
Obviously this value is "unique" in terms of the treatment the person gives it, not because it is identified by an internal value of the device that generates the signal. There is no possibility of accessing an identifier of type 'serial_number' or 'mac_address' from the RF signal.

But it is also true that this implementation enables the possibility of this operation being perverted so that you can substitute one physical device for another and assign it the same `device_id`.

So before going any further in this RP I think you should evaluate if the approach is correct. For my part, I don't think there is another way to generate a unique_id value for this type of integration.

bdraco committed:
Update HAP-python to 2.8.4 (#35541)

Fix race condition that causes pairing and unpairing failures. ikalchev/HAP-python#246
Fix loop on dropped connections that causes temporary stalls and connection loss. ikalchev/HAP-python#249
Fix exception on missing video fields. ikalchev/HAP-python#245

bdraco committed:
Fix sense doing I/O in the event loop (#35523)

When the websocket is created `SSLContext.load_default_certs` is called
each time which opens up the system default ssl certs file and reads it in

Normally this goes unnoticed, however since there are frequent connects
and disconnects of the websocket it was noticeable.

balloob committed:
Add email as common data string (#35546)

balloob committed:
Add temperature device class to Darksky sensors (#35544)

balloob committed:
Use config flow references for data (#35529)

marcelveldt committed:
Add binary sensor platform to zwave_mqtt (#35519)

* add binary sensor platform to zwave_mqtt

* add tests for binary_sensor

* fix tests

* device class is required value

* Update homeassistant/components/zwave_mqtt/binary_sensor.py

use colon as separator

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* extend tests

* code optimize

* add test for enabling a legacy binary sensor

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Danielhiversen committed:
Upgrade opengarage library (#35549)

rajlaud committed:
Fix examples in services.yaml for squeezebox component (#35548)

liudger committed:
Bump bsblan lib to version 0.3.7 (#35551)

bdraco committed:
Ensure homekit_controller recieves zeroconf c# updates (#35545)

* Ensure homekit_controller recieves zeroconf c# updates

If an integration has a homekit config flow step
homekit controller would not see updates for
devices that were paired with it and would not
rescan for changes.

* Only send updates to homekit controller if the device is paired

This avoids the device showing up a second time.

* remove debug

* fix refactor error

OnFreund committed:
Try to automatically detect zones on first run of Monoprice integration (#35127)

jtommi committed:
Implement soundtouch select source (#31669)

bramkragten committed:
Updated frontend to 20200512.0 (#35558)

emlove committed:
Add Zerproc integration (#35477)

frenck committed:
Add more system information from Supervisor (#35560)

Kane610 committed:
UniFi - Sites don't declare role on UniFiOS 1.7.0 beta (#35555)

* New way to identify role, compatible with standalone controller or as part of unifios

* Remove error

* Bump dependency to v22

* Remove unused import

shenxn committed:
Fix synology temperature rounding (#35556)

* Fix synology temperature rounding

* Remove stale comment

jjlawren committed:
Bump plexapi to 3.6.0 (#35563)

MartinHjelmare committed:
Patch aiohttp client session close (#34769)

* Patch aiohttp client session close

* Add test

* Restore close regardless of auto_cleanup

* Close session instead of detaching and do not restore

* Delint test

* Add frame helper

* Use frame helper

* Test warning log when closing session

* Clean up

* Correct docstring

* Do not change shutdown

* Fix tests

felipediel committed:
Fix connection problems in the Broadlink integration (#34670)

* Use helper functions for exception handling

* Create a separate class to handle communication with the device

* Update manifest

* Use coroutine for service setup

* Fix sensor update

* Update tests

* Fix MP1 switch

* Add device.py to .coveragerc

* Remove unnecessary blocking from test_learn_timeout

* Change access method for entries with default values

* Make the changes suggested by MartinHjelmare

* Remove dot from debug message

* Use underscore for unused variable

mampfes committed:
Add wiffi integration (#30784)

* Add integration for wiffi devices

wiffi devices are DIY board manufactured by stall.biz.
Several devices are available, e.g. a weather station (weatherman), an
indoor environmental sensor (wiffi-wz) and some more.
This intgration has been developed using a weatherman device, but should
also work for other devices from stall.biz.

* Fix pylint warning

* Use WIFFI / STALL WIFFI instead of wiffi to be consistent with stall.biz

* Don't update disabled entities.

* fix complains

- move wiffi specific code to pypi
- remove yaml configuration code

* incorporate various suggestions from code review

* fix remaining comments from Martin

* fix comments

* add tests for config flow

* fix comments

* add missing requirements for tests

* fix pylint warnings

* fix comments

* fix comments

remove debug log
rename .translations to translations

* rebase and adapt to latest dev branch

* Update homeassistant/components/wiffi/config_flow.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/wiffi/config_flow.py

Co-Authored-By: Martin Hjelmare <marhje52@gmail.com>

* fix missing import

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

jjlawren committed:
Plex discovery on demand (#35303)

* Allow Plex discovery on demand

* Add new discovery source

* Add tests

uvjustin committed:
Add forked_daapd integration (#31953)

* New forked_daapd component

* Bunch of changes

Add config flow and zeroconf
Add zones on callback when added by server
Add password auth
Add async_play_media for TTS
Add media_image_url
Add support for pipe control/input from librespot-java
Improve update callbacks

* Refactor as per code review suggestions

Move config_flow connection testing to pypi library (v0.1.4)
Remove use of ForkedDaapdData class
Decouple Master and Zone data and functions
Add updater class to manage websocket and entity updates

* More changes as per code review

Avoid direct access to entities in tests
Bump pypi version
Mark entities unavailable when websocket disconnected
Move config tests to test_config_flow
Move full url creation from media_image_url to library
Move updater entity from master to hass.data
Remove default unmute volume option
Remove name from config_flow
Remove storage of entities in hass.data
Use async_write_ha_state
Use signal to trigger update_options
Use unittest.mock instead of asynctest.mock

* Yet more changes as per code review

Add more assertions in tests
Avoid patching asyncio
Make off state require player state stopped
Only send update to existing zones
Split up some tests
Use events instead of async_block_till_done
Use sets instead of lists where applicable
Wait for pause callback before continuing TTS

* Remove unnecessary use of Future()

* Add pipes and playlists as sources

* Add support for multiple servers

Change config options to add max_playlists+remove use_pipe_control
Create Machine ID in test_connection and also get from zeroconf
Modify hass.data storage
Update host for known configurations
Use Machine ID in unique_ids, entity names, config title, signals

* Use entry_id as basis for multiple entries

* Use f-strings and str.format, abort for same host

* Clean up check for same host

farmio committed:
update knx module to xknx 0.11.3 (#35154)

- add expose attribute function and default value
- default climate havc_mode to "heat" if modes are not supported
- support `update_entity` service call

shenxn committed:
Add config flow to pi_hole integration (#35442)

* Add config flow to pi-hole

* Add config flow tests

* Change PlatformNotReady to ConfigEntryNotReady

* Improve config flow

* Add @shenxn as codeowner

* Use entity_id as unique id

* Remove .get with [] for required fields

* Remove unique id from config flow

* Replace some strings with references

* Fix api_key string

* Fix service api_key check

* Remove unused DuplicatedNameException

bramkragten committed:
Updated frontend to 20200513.0 (#35574)

fronzbot committed:
Add config flow and 2FA support for Blink (#35396)

bdraco committed:
Add ssdp discovery for isy994 (#35568)

* Add ssdp discovery for isy994

* Increase test coverage for existing config flow

* Update tests/components/isy994/test_config_flow.py

Co-authored-by: shbatm <support@shbatm.com>

* Update tests/components/isy994/test_config_flow.py

Co-authored-by: shbatm <support@shbatm.com>

* move constants

* Update tests/components/isy994/test_config_flow.py

Co-authored-by: shbatm <support@shbatm.com>

* undo CONF_TLS_VER from homeassistant.const

Co-authored-by: shbatm <support@shbatm.com>

frenck committed:
Merge branch 'master' into dev

balloob committed:
Update translations

frenck committed:
Merge branch 'dev' into rc

frenck committed:
Bumped version to 0.110.0b0

frenck committed:
Bumped version to 0.111.0dev0 (#35593)

scop committed:
Upgrade flake8 to 3.8.1, fix findings (#35578)

homeassistant committed:
[ci skip] Translation update

bramkragten committed:
Updated frontend to 20200514.0 (#35598)

bramkragten committed:
Updated frontend to 20200514.0 (#35598)

hunterjm committed:
Additional checks for ONVIF event capabilities (#35599)

catch any exceptions when pulling event capabilities and assume it is not supported

hunterjm committed:
Additional checks for ONVIF event capabilities (#35599)

catch any exceptions when pulling event capabilities and assume it is not supported

bouwew committed:
Bump haanna to 0.15.0 (#35579)

* Link to haanna v0.15.0

* Update requirements_all.txt

bouwew committed:
Bump haanna to 0.15.0 (#35579)

* Link to haanna v0.15.0

* Update requirements_all.txt

scop committed:
Run pre-commit gen_requirements_all on pre-commit config changes (#35588)

Kane610 committed:
Axis - Code improvements (#35592)

* Adapt library improvements
Clean up integration and tests and make them more like latest changes in UniFi integration

* Bump dependency to v26

bachya committed:
Remove Automatic integration (#35029)

zacpotts committed:
Fix zwave thermostat specific device type (#35609)

zacpotts committed:
Fix zwave thermostat specific device type (#35609)

frenck committed:
Rename zigbee to xbee (#35613)

starkillerOG committed:
Xiaomi Miio zeroconf discovery (#35352)

marcelveldt committed:
Fix zwave_mqtt creating the device name (#35603)

* Fix for creating the device name

Creating the devicename included a typo and was missing the custom (preferred) name set by the OZW Admin tool.

* update comments

marcelveldt committed:
Fix zwave_mqtt creating the device name (#35603)

* Fix for creating the device name

Creating the devicename included a typo and was missing the custom (preferred) name set by the OZW Admin tool.

* update comments

scop committed:
Upgrade huawei-lte-api to 1.4.12 (#35618)

https://github.com/Salamek/huawei-lte-api/releases/tag/1.4.12

scop committed:
Upgrade huawei-lte-api to 1.4.12 (#35618)

https://github.com/Salamek/huawei-lte-api/releases/tag/1.4.12

amelchio committed:
Upgrade to pysonos 0.0.29 (#35617)

amelchio committed:
Upgrade to pysonos 0.0.29 (#35617)

hunterjm committed:
additional log info and strings fix (#35622)

hunterjm committed:
additional log info and strings fix (#35622)

bdraco committed:
Fix reversed logic in zeroconf homekit pairing check (#35596)

* Fix reversed logic in zeroconf homekit pairing check

* s/server_info/service_info/

bdraco committed:
Fix reversed logic in zeroconf homekit pairing check (#35596)

* Fix reversed logic in zeroconf homekit pairing check

* s/server_info/service_info/

bramkragten committed:
Add check for HTML in translations (#35615)

* Add check for HTML in translations

and remove existing html

* Add test

bramkragten committed:
Add check for HTML in translations (#35615)

* Add check for HTML in translations

and remove existing html

* Add test

uvjustin committed:
Clean up forked_daapd volume saving/setting in async_play_media (#35584)

* Clean up volume saving/setting in async_play_media

* Set source to pipe when queued externally

* Add server version requirement to error string

uvjustin committed:
Clean up forked_daapd volume saving/setting in async_play_media (#35584)

* Clean up volume saving/setting in async_play_media

* Set source to pipe when queued externally

* Add server version requirement to error string

zewelor committed:
Provide yeelight unique_id using ssdp discovery (#35448)

* Provide yeelight unique_id using ssdp discovery

* Fixes

* Comment fix

* Cleanup initialization logic and add unique id to binary sensor

* Update homeassistant/components/yeelight/__init__.py

Co-authored-by: Teemu R. <tpr@iki.fi>

* Update homeassistant/components/yeelight/__init__.py

Co-authored-by: Teemu R. <tpr@iki.fi>

* Update comment

* Update comment

* Fix wrong model docstring

Co-authored-by: Teemu R. <tpr@iki.fi>

Adminiuga committed:
Don't remove deprecated ZHA config option yet (#35627)

Adminiuga committed:
Don't remove deprecated ZHA config option yet (#35627)

bramkragten committed:
Updated frontend to 20200514.1 (#35632)

bramkragten committed:
Updated frontend to 20200514.1 (#35632)

isk0001y committed:
Add tado zone variable open window detected (#34969)

* openwindow0: this is the part for the HA itself

* Update homeassistant/components/tado/sensor.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Update homeassistant/components/tado/sensor.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* new version after review

Co-authored-by: J. Nick Koston <nick@koston.org>

frenck committed:
Rename zwave_mqtt to ozw (#35631)

frenck committed:
Rename zwave_mqtt to ozw (#35631)

StevenLooman committed:
Properly handle incomplete upnp ssdp discovery (#35553)

StevenLooman committed:
Properly handle incomplete upnp ssdp discovery (#35553)

bskaplou committed:
Add icons for upload/download sensors for asuswrt (#35581)

elupus committed:
Add device triggers to support setting turn_on event (#35456)

* Add device triggers to support setting turn_on event

* Add turn on event

* Add unique_id based on config entry

* Adjust tests for addition of uuid

* Supported features are now always same

* Switch to player_setup fixture that actually start platform

* Update homeassistant/components/arcam_fmj/const.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

danielperna84 committed:
Fix HM-CC-VD unit, #31083 (#35066)

frenck committed:
Update translations for OZW and ONVIF

frenck committed:
Bumped version to 0.110.0b1

bdraco committed:
Fix CI failure on arcam tests (#35641)

homeassistant committed:
[ci skip] Translation update

Quentame committed:
Bump python-synology to 0.8.1 (#35640)

* Bump python-synology to 0.8.1

* Fix tests

Quentame committed:
Bump python-synology to 0.8.1 (#35640)

* Bump python-synology to 0.8.1

* Fix tests

bramkragten committed:
Bumped version to 0.110.0b2

shenxn committed:
Bump roombapy to 1.6.1 (#35650)

* Bump roombapy to 1.6.1

* Improve roomba error handling

shenxn committed:
Bump roombapy to 1.6.1 (#35650)

* Bump roombapy to 1.6.1

* Improve roomba error handling

bramkragten committed:
Fix caldav event for calendar panel (#35653)

bramkragten committed:
Fix caldav event for calendar panel (#35653)

frenck committed:
Add more information to discovery API (#35624)

zewelor committed:
Add explicit return none and binary sensor availability to yeelight (#35649)

* Add explicit return none and binary sensor availbility

* Fix

bdraco committed:
Update cast to use shared zeroconf (#35570)

* Update cast to use the shared zeroconf instance

* Add zeroconf to after_dependencies

* Bump version to 5.2.0

hunterjm committed:
ONVIF: Add check around media capabilities (#35667)

hunterjm committed:
ONVIF: Add check around media capabilities (#35667)

Kane610 committed:
UniFi - Remove unused string definition, pointed out by Bram (#35678)

gadgetmobile committed:
Add Blebox switch support (#35371)

* support BleBox switches

* fix tox py37 test failures

* refactor BleBox device class map

Kane610 committed:
Axis - Streamline setup and teardown of integration (#35675)

* Streamline setup and teardown of integration

* Dont remove integration twice

bramkragten committed:
Updated frontend to 20200515.0 (#35677)

bramkragten committed:
Updated frontend to 20200515.0 (#35677)

homeassistant committed:
[ci skip] Translation update

uvjustin committed:
Change MediaPlayerDevice to MediaPlayerEntity (#35692)

uvjustin committed:
Change MediaPlayerDevice to MediaPlayerEntity (#35692)

Commit:
Update Universal Powerline Bus event name (#35644)

Commit:
Update Universal Powerline Bus event name (#35644)

ctalkington committed:
Prevent discovery of IPP printers lacking identifier (#35630)

ctalkington committed:
Prevent discovery of IPP printers lacking identifier (#35630)

frenck committed:
Bumped version to 0.110.0b3

frenck committed:
Fix handling of additional data in core config storage (#35660)

frenck committed:
Fix handling of additional data in core config storage (#35660)

bdraco committed:
Use built in queue log handlers to avoid formatting logs in the event loop (#35633)

* Use built in queue log handlers to avoid formatting logs in the event loop

Logging is now formatted and written in another thread to
ensure there is minimal impact on the event loop when
a log message is processed.

This change replaces the existing AsyncHandler log handler
as python 3.7+ now offers an off the shelf solution

* add a simple test

* s/async_migrate_log_handlers_to_queue/async_activate_log_queue_handler/g

gadgetmobile committed:
Add Blebox air quality support (#35372)

* support BleBox air-quality

* fixed switch tests via cherry-pick from #35552

* fix test after cherry-picking

* fix flake8 issues

vangorra committed:
Add config flow to gogogate2 component (#34709)

* Add config flow to gogogate2 component.

* Using a more stable gogogate api.

* Getting config flows working better by using different downstream library.

* Fixing options not getting default values.
Adding availability to cover entity.

* Simplifying return types of function.

* Address PR feedback.

* Making user config flow not abort.

* Using DataUpdateCoordinator.

* Addressing PR feedback.

* Using standard method for using hass.data

* Split auth fail test into separate tests.

danielperna84 committed:
Update pyhomematic to 0.1.67 (#35637)

Adminiuga committed:
Bump up ZHA dependencies (#35706)

Adminiuga committed:
Bump up ZHA dependencies (#35706)

bdraco committed:
Prevent updater from delaying startup (#35708)

* Prevent updater from delaying startup

The updater sometimes times out as seen in
#33833 and the linked issues. The issue was
presenting again today as it appears the service
is overloaded again.

* s/hass.loop/asyncio/g

homeassistant committed:
[ci skip] Translation update

cgiraldo committed:
deCONZ - Add support for "Window covering controller" (#35294)

* Add support for "Window covering controller" 

Some window cover devices such as Sunricher SR-ZG9080A are identified in deconz as "Window covering controller".
We need to include this device type in const.py to be integrated as type "cover" in home-assistant.

* Added Window covering controller to deconz tests

mampfes committed:
Add config option to set timeout for wiffi devices (#35694)

* add config option to set timeout for wiffi devices

Wiffi devices allow to configure the update period (=
full_loop_minutes). The integration shall respect the configured update
period and therefore need a configuration for the timeout, too.

* Move timeout from config flow to option flow

* add test for option flow

jfmcarreira committed:
Add support for custom media_type in mediaroom (#34625)

* Add support for custom media_type in mediaroom

* Clean logging

* Fix formatting

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

elupus committed:
Support config entry unload in arcam_fmj (#35656)

* Support entry unload

* Switch to create_task and skip checking for available task

Danielhiversen committed:
Upgrade opengarage lib to 0.1.4 (#35729)

Danielhiversen committed:
Upgrade opengarage lib to 0.1.4 (#35729)

raman325 committed:
Reduce Vizio API calls (#35726)

* reduce calls to API

* get apps list every time

* fix typo

atmurray committed:
Add Acmeda integration (#33384)

* First cut of Rollease Acmeda Pulse Hub integration.

* Acmeda integration improvements:
 - Moved common code into a base entity
 - Battery level sensor added
 - Localisation now working

* Added requirement for aiopulse now that it has been uploaded to PyPI.

* Exclude acmeda integration from coverage check as it relies on a hub being present.

* Fix Travis CI build issues.

* Remove unused constants.

* Remove unused group logic from cover.py

* Removed commented code from base.py

* Remove sensors (battery entities) on removal of hub.

* Remove unused groups from sensor.py

* Acmeda device and entity update made fully asynchronous using subscriptions to remove need for config polling.

* Updated aiopulse version dependency.
Removed non-functional battery charging indication.

* Rationalised common code to update entities into helpers.py

* Fix linting issue.

* Correct additional CI pylint errors.

* Index config_entries by entry_id.
Move entity loading and unloading to __init__.py
Add entry_id to dispatcher signal
Removed now unused polling code hub
Added config_flow unit tests

* Tweak to integration config_entry title.

* Bumped aiopulse module to 0.3.2.
Reduced verbosity of aiopulse module.

* Changed to using direct write of device state.
Removed old style async_step_init config_flow step.

* Remove superfluous battery_level and device_state_attributes from battery entity.

* Removal of unused strings.
Removal of unused create_config_flow helper.
Removal of stale comment.

* Remove use of shared container to track existing enities.
Moved removal and deregistration of entities to base class through use of dispatch helper.

* Fixed strings.json

* Fix incorrect use of remove instead of pop on dict.

* Add support for tilting covers, bump aiopulse version number.

* Bump aiopulse version to v0.3.4.
Fixed bug in cover supported_features.

* Bumped aiopulse version to 0.4.0
Update acmeda .coveragerc exclusions

* Removed already configured hub check from __init__.py async_setup_entry
Removed passing in hass reference to base entity class
Renamed entity async_reset to async_will_remove_from_hass
Changed device_info and properties
Migrated to CoveEntity from CoverDevice
Added dispatched_connect cleanup on hub removal
Removed unused entries from manifest
Removed override of battery icon
Renamed translations folder

* Reversed unintended change to .coveragerc

* Fixed config flow for multi-hub discovery.

* Acmeda enhancements as requested by MartinHjelmare

* Force import to connect to hub to retrieve id prior to creating entry

* Remove YAML configuration support.

* Tidied up config_flow and tests:
 - removed unnecessary steps
 - fixed typos

* Removed storage of hub in config_flow.

bdraco committed:
Avoid locking in the logging queue handler (#35700)

* Avoid locking in the logging queue handler

We do not need a lock here as the underlying queue is already
thread safe.

* Add coverage for logging handle

shenxn committed:
Fix garmin_connect test (#35724)

rhadamantys committed:
Add restore last state for EnOcean Sensors (#34375)

teharris1 committed:
Add .env file to .gitignore for vscode environment settings (#35200)

uvjustin committed:
Abort forked-daapd zeroconf flow if version < 27 (#35709)

* Change MediaPlayerDevice to MediaPlayerEntity

* Abort zeroconf if mtd-version < 27.0

uvjustin committed:
Abort forked-daapd zeroconf flow if version < 27 (#35709)

* Change MediaPlayerDevice to MediaPlayerEntity

* Abort zeroconf if mtd-version < 27.0

teharris1 committed:
Change Insteon backend module to pyinsteon from insteonplm (#35198)

* Migrate to pyinsteon from insteonplm

* Rename devices entities

* Print ALDB even if not loaded

* Add relay to name map

* Change insteonplm to pyinsteon

* Update requirements_all correctly

* Code review updates

* async_set_speed receive std speed value

* default speed to std medium value

* Call async methods for fan on/off

* Comment await required in loop

* Remove emtpy and add codeowner

* Make services async and remove async_add_job call

* Remove extra logging

* New device as async task and aldb load in loop

* Place lock in context bloxk

* Limiting lock to min

* Remove .env file

springstan committed:
Add and use ELECTRICAL_CURRENT_AMPERE, ELECTRICAL_VOLTAGE_AMPERE constants (#33990)

* Add and use ELECTRICAL_CURRENT_AMPERE constant

* Add and use ELECTRICAL_VOLTAGE_AMPERE constant

* Rename ELECTRICAL_VOLTAGE_AMPERE to ELECTRICAL_VOLT_AMPERE

* Fix imports

sqldiablo committed:
Bump PyWeMo version to 0.4.43 (#35693)

* PyWeMo version bump to 0.4.43

Changes necessary to safely upgrade to pyWeMo 0.4.43. This includes catching ActionExceptions that may be thrown by pyWeMo when it is unable to interact with a physical device.

* Black formatting fix

* Fix isort issues

* Code review changes

* More code review fixes

* Linting fix

* Undo dict.get change

* Change a couple instances of dict[key] to dict.get

bdraco committed:
Ensure homekit version strings conform to spec (#35741)

HomeKit requires all version strings to be in the
format MAJOR.MINOR.REVISION

bdraco committed:
Ensure homekit version strings conform to spec (#35741)

HomeKit requires all version strings to be in the
format MAJOR.MINOR.REVISION

tetienne committed:
Add Somfy hub as device (#35209)

bdraco committed:
Set lifx device sw_version (#35739)

This change solves an issue where homekit would
think the firmware was out of date on the bulb
because no version was set.

bdraco committed:
Add sw_version and connections to sonos devices (#35743)

PlasmaEye committed:
Add ARWN rain total and rain rate sensors (#35751)

* add rain total and rain rate sensors

* linting

* linting

gadgetmobile committed:
Add Blebox lights support (#35370)

* add BleBox lights support

* cherry pick refactoring from #35552

* Inherit from LightEntity instead of Light

Co-authored-by: J. Nick Koston <nick@koston.org>

* import LightEntity instead of Light

Co-authored-by: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

gadgetmobile committed:
Cleanup BleBox platforms (#35552)

homeassistant committed:
[ci skip] Translation update

BKPepe committed:
Upgrade sqlalchemy to 1.3.17 (#35745)

amelchio committed:
Handle Sonos changing IP address (#35639)

Danielhiversen committed:
Update mill manifest to reflect config flow (#35748)

Danielhiversen committed:
Update mill manifest to reflect config flow (#35748)

bdraco committed:
Handle UPS disconnects in NUT (#35758)

bdraco committed:
Handle UPS disconnects in NUT (#35758)

fredrike committed:
Fix daikin discovery flow (#35767)

fredrike committed:
Fix daikin discovery flow (#35767)

MatsNl committed:
Bump Atag dependency to 0.3.1.2 (#35776)

MatsNl committed:
Bump Atag dependency to 0.3.1.2 (#35776)

bdraco committed:
Avoid a context switch in the history api (#35716)

* Avoid a context switch in the history api

The history api was creating a job to fetch the
states and another job to convert the states to
json. This can be done in a single job which
decreases the overhead of the operation.

* Revert to original solution to avoid function redefine each call

bachya committed:
Add support for templates in Slack blocks (#34704)

bramkragten committed:
Updated frontend to 20200518.0 (#35785)

bramkragten committed:
Updated frontend to 20200518.0 (#35785)

uvjustin committed:
Skip forked_daapd ignored entries with empty entry.data (#35772)

uvjustin committed:
Skip forked_daapd ignored entries with empty entry.data (#35772)

gadgetmobile committed:
Fix review requests for BleBox support from #35370 (#35786)

* fix review requests from #35370

* fix pylint W0621 (redefined-outer-name)

frenck committed:
Bumped version to 0.110.0b4

homeassistant committed:
[ci skip] Translation update

hunterjm committed:
Fix ONVIF subscription renewal (#35792)

* fix subscription renewal

* catch ValueError for #35762

hunterjm committed:
Fix ONVIF subscription renewal (#35792)

* fix subscription renewal

* catch ValueError for #35762

bdraco committed:
Avoid creating multiple sqlalchemy sessions in a single history call (#35721)

* Avoid a context switch in the history api

The history api was creating a job to fetch the
states and another job to convert the states to
json. This can be done in a single job which
decreases the overhead of the operation.

* Ensure there is only one sqlalchemy session created per history
query.

Most queries created three sqlalchemy sessions which was
especially slow with sqlite since it opens and closes the
database.

In testing the UI is noticeably faster at generating history
graphs for entites.

* Add additional coverage

* pass hass first to _states_to_json and _get_significant_states

amelchio committed:
Upgrade pysonos to 0.0.30 (#35793)

amelchio committed:
Upgrade pysonos to 0.0.30 (#35793)

Adminiuga committed:
Bump up ZHA dependencies. (#35797)

Adminiuga committed:
Bump up ZHA dependencies. (#35797)

cyberjunky committed:
Update to garminconnect 0.1.13 (#35750)

* Update to garminconnect 0.1.13

This will fix body composition sensors being unavailable

* Update requirements_all to garminconnect 0.1.13

* Update requirements_test_all.txt

felipediel committed:
Allow underscores in broadlink hostnames (#35791)

* Allow underscores in hostnames

* Ignore case

nickw444 committed:
Fire events on homekit TV remote key press (#29588)

* Fire events on homekit TV remote key press

* Changes from code review

* black

* isort

* flake8

* Update tests/components/homekit/test_type_media_players.py

Co-authored-by: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

uvjustin committed:
Change version check in forked-daapd zeroconf step (#35796)

uvjustin committed:
Change version check in forked-daapd zeroconf step (#35796)

bramkragten committed:
Updated frontend to 20200519.0 (#35813)

bramkragten committed:
Updated frontend to 20200519.0 (#35813)

MartinHjelmare committed:
Bump codecov to 2.1.0 (#35808)

* Bump codecov to 2.1.0

* Try to invalidate cache

odinuge committed:
Fix timezone issues for db fields in recorder (#35719)

The database fields are timezoned via DateTime(timezone=True), so the
default value should be timezoned too. When using cockroachdb this is
fatal and results in the recorder crashing.

teharris1 committed:
Fix Insteon cover and smoke detector entities (#35810)

* Bump pyinsteon to 1.0.1

* Fix cover open/close calls

* Add smokebridge sensors

* trigger tests

* trigger tests

bramkragten committed:
Bumped version to 0.110.0b5

Kane610 committed:
UniFi - Fix disabled entities being enabled after a restart (#35819)

* Async remove call removed too much, resulting in disabled entities coming back after a restart

* Calling super().async_remove is no longer needed, changed to self.async_remove

* Yes, they should be sets...

Kane610 committed:
UniFi - Fix disabled entities being enabled after a restart (#35819)

* Async remove call removed too much, resulting in disabled entities coming back after a restart

* Calling super().async_remove is no longer needed, changed to self.async_remove

* Yes, they should be sets...

bskaplou committed:
Add xiaomi miio services remote_set_led_on/off (#35805)

* Add services remote_set_led_on/remote_set_led_off to control changmi_ir led

* pylint: elif => if

* Fix services.async_register => platform.async_register_entity_service

* Update homeassistant/components/xiaomi_miio/remote.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* more fixes

* async

* more fixes

* fix

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

homeassistant committed:
[ci skip] Translation update

shenxn committed:
Add dynamic icon to roomba battery sensor (#35647)

* Add dynamic icon to roomba battery sensor

* Fix lint

* Rename _state to _robot_state

frenck committed:
Add UUID to ZeroConf service info (#35623)

frenck committed:
Bumped version to 0.110.0

frenck committed:
Merge pull request #35828 from home-assistant/rc

frenck committed:
Merge branch 'master' into dev

fredrike committed:
Fix Daikin duplicate entries (#35833)

thomkaufmann committed:
Add Nuki Opener integration (#35702)

* Add Nuki Opener integration

* Update pynuki version requirement; fix typo

* Update requirements_all.txt

* Create base class of shared lock and opener code

* Clean up code formatting

* Update requirements_all; Run isort

* Remove unnecessary pass statements

Emilv2 committed:
Fix Delijn sensor naming (#35789)

uvjustin committed:
Handle None received from pyforked-daapd (#35830)

* Handle None received from API in forked-daapd

* Bump pyforked-daapd version in requirements

* Add test

Adminiuga committed:
Add zha climate (#35682)

* Initial ZHA climate implementation.
* Add retryable_request decorator.
sort helpers imports.

* Check manufacturer for Climate implementation.
* Default zha.climate.operation_list to [Off]
* Migrate to climate 1.0
* Sort imports, properties and methods.
* Handle 'PRESET_NONE' presets.
* Use pi_heating/cooling_demand for HVAC action prop.
* Implement `running_state` HVAC channel property.
For ZHA thermostats which don't support `pi_heating_demand` or
`pi_cooling_demand' attributes.

* wip fan support
* Refactor retryable request logging.
* Rebase cleanup.
* Update climate discovery.
* Fix ZHA climate restoration.
* Bulk configure attribute reports.
* Use configure_reporting_multiple command for Light
More detailed response parsing of configure_reporting_multiple.

* Use ordered list for HVAC cluster attribute reports.
* Don't mutilate HVAC mode list.
* Add fan_mode property to fan channel.
* Fix type hinting.
* Expose fan mode only.
* Implement fan mode setting.
Drop support for HVAC_FAN_ONLY mode.

* Use ClimateEntity as base class.
* Cleanup debug code.
* Update time display for Sinope.
* Don't do many retries.
* Don't use multi attr reporting configuration.
* Make tests pass.
* Drop support for setpoint change source/amount.
* Cleanups.
* Drop aux heat
* Update tests.
* Drop Sinope temperature display code.
* Update tests.
* Refactor temperature setting.
* Update tests.
* Update Fan tests.
* Lint
* Black.
* Use correct logging levels

MartinHjelmare committed:
Add climate services required features (#35804)

bachya committed:
Fix Prezzibenzina doing I/O in the event loop (#35881)

* Fix Prezzibenzina doing I/O in the event loop

* Linting

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Fix zha climate tests (#35893)

hunterjm committed:
fix mjpeg issue along with some cameras not returning event capabilities properly (#35885)

bdraco committed:
Ensure http can startup if homekit fails to load (#35888)

* Ensure HomeAssistant can startup if homekit fails to load

* Update homeassistant/components/logbook/manifest.json

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Fix legacy Hunter Douglas PowerView devices (#35895)

These devices are missing firmware information
as the 1.0 firmware did not provide it.

ctalkington committed:
Update rokuecp to 0.4.1 (#35899)

* update rokuecp to 0.4.1

* Update requirements_all.txt

* Update requirements_test_all.txt

bramkragten committed:
Updated frontend to 20200519.1 (#35877)

bdraco committed:
Ensure storage write consume the data under the lock (#35889)

If two writes trigger at the same time the data would already
be consumed.

bdraco committed:
Homekit should skip devices that are missing in device registry (#35857)

* Homekit should skip devices that are missing in device registry

* Add test for this failure state

danielperna84 committed:
Fix light profiles for HomeMatic lights (#35882)

bachya committed:
Auto-level AirVisual API calls (#34903)

marcelveldt committed:
Add climate platform to ozw (#35566)

bramkragten committed:
Update frontend to 20200519.3 (#35925)

* Updated frontend to 20200519.2

* Updated frontend to 20200519.3

homeassistant committed:
[ci skip] Translation update

fredrike committed:
Fix Daikin AC integration for AirBase units (#35952)

Commit:
Fix typo in conf[CONF_SSL] (#35946)

Adminiuga committed:
No side effects in zha climate properties (#35942)

* No side effects in properties

* Remove logging

hunterjm committed:
Bump pyAV and close unclosed outputs (#35960)

* bump pyAV and close unclosed outputs

* skip stream from coverage for now

* fix divide by zero error

bramkragten committed:
Update frontend to 20200519.4 (#35987)

jhollowe committed:
Update proxmoxer to 1.1.0 (#35951)

MartinHjelmare committed:
Bump python-openzwave-mqtt to 1.0.2 (#35980)

MartinHjelmare committed:
Fix lutron_caseta setup options (#35974)

hunterjm committed:
Fix ONVIF Transport (#35932)

* allow lib to create AsyncTransport

* fix transport close issue

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Ensure homekit functions if numpy is unavailable (#35931)

hunterjm committed:
Fix ONVIF PTZ and profile encoding issues (#36006)

* allow lib to create AsyncTransport

* fix transport close issue

* fix zoom only cameras without PTZ presets

* catch profiles without encoding configuration

* also catch ServerDisconnectedError for ptz

flz committed:
Fix iaqualink sensors (#36000)

* iaqualink: small sensor fixes

* Re-add device_class, fix type hints.

flz committed:
Bump iaqualink to 0.3.3 (#35999)

alengwenus committed:
Fix device_registry cleanup behavior (#35977)

* Fix: Only decives which are not referenced by an entity or a config_entry are removed

* Adapted test for async_cleanup

* Changed variable names

ctalkington committed:
Fix roku play/pause support (#35991)

amelchio committed:
Update sonos codeowners (#36016)

fredrike committed:
Deprecate Daikin yaml-support (#35768)

felipediel committed:
Handle StorageError in the Broadlink integration (#35986)

Adminiuga committed:
Fix ZHA climate hvac_action for Centralite thermostat (#35993)

* Centralite specific control seq of operation

* Remove Fan safeguards

* Split hvac_action property.

* Refactor hvac_action property.

Current hvac_action logic is Zen Within thermostat specific and differs
a bit from ZCL specs. Implement it as a separate class.

* Refactor hvac_action property for default thermostat

Follow more closely ZCL specs in parsing hvac state of the thermostat.

soldag committed:
Migrate rpi_gpio_pwm to extend LightEntity instead of Light (#36028)

bdraco committed:
Use a single service browser for zeroconf discovery (#35997)

chmielowiec committed:
Fix service registration supported features check (#35718)

bdraco committed:
Ensure configured logger severity is respected (#35749)

emontnemery committed:
Reset zeroconf log level (#36002)

bdraco committed:
Fix shade compatibility with hunter douglas powerview 1.0 hubs (#36040)

frenck committed:
Upgrade hass-nabucasa to 0.34.3 (#36025)

Jc2k committed:
Bump aiohomekit (#36041)

ctalkington committed:
Improve ipp unique id parsing (#35959)

mnaggatz committed:
Add support for Velux Gates (#34774)

fredrike committed:
Bump tellduslive version (#36048)

michaeldavie committed:
Update environment_canada camera (#36010)

* Bump env_canada to 0.0.38

* Fix timestamp type

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Add ZHA roller shadows as cover instead of switch (#36059)

* Implement cover for "Shade" ZHA device type.
* Update ZHA cover tests.
* Add stop command
* Coverage.

basnijholt committed:
Use Coerce(float) on service options for kef (#35659)

* use Coerce(float) on service options

* defer the type from options list

javicalle committed:
Merge branch 'rflink_unique_id' of https://github.com/javicalle/home-assistant into rflink_unique_id

javicalle commented:
It seems that I have done a mess with code rebase.
Also, there is no interest or activity about it, so I close the PR

jjansen85 commented:
> It seems that I have done a mess with code rebase.
> Also, there is no interest or activity about it, so I close the PR

That's a pity, that's exactly what I was looking for. Any chance this can still be implemented?

MuadDib007 commented:
unique_id would make my life much easier with regards to RFLink so I would like to see it as well.
